tests/aggregation/tests.py

```diff
@@ -1402,3 +1420,199 @@ def test_coalesced_empty_result_set(self):
                 )['latest_opening'],
                 datetime.datetime,
             )
+
+    def test_aggregation_default_unsupported_by_count(self):
+        msg = 'Count does not allow default.'
+        with self.assertRaisesMessage(TypeError, msg):
+            Count('age', default=0)
+
+    def test_aggregation_default_unset(self):
+        for Aggregate in [Avg, Max, Min, StdDev, Sum, Variance]:
+            with self.subTest(Aggregate):
+                result = (
+                    Author.objects.filter(age__gt=100)
+                    .aggregate(value=Aggregate('age'))
+                )
+                self.assertIsNone(result['value'])
+
+    def test_aggregation_default_zero(self):
+        for Aggregate in [Avg, Max, Min, StdDev, Sum, Variance]:
+            with self.subTest(Aggregate):
+                result = (
+                    Author.objects.filter(age__gt=100)
+                    .aggregate(value=Aggregate('age', default=0))
+                )
+                self.assertEqual(result['value'], 0)
+
+    def test_aggregation_default_integer(self):
+        for Aggregate in [Avg, Max, Min, StdDev, Sum, Variance]:
+            with self.subTest(Aggregate):
+                result = (
+                    Author.objects.filter(age__gt=100)
+                    .aggregate(value=Aggregate('age', default=21))
+                )
+                self.assertEqual(result['value'], 21)
+
+    def test_aggregation_default_expression(self):
+        for Aggregate in [Avg, Max, Min, StdDev, Sum, Variance]:
+            with self.subTest(Aggregate):
+                result = (
+                    Author.objects.filter(age__gt=100)
+                    .aggregate(value=Aggregate('age', default=Value(5) * Value(7)))
+                )
+                self.assertEqual(result['value'], 35)
+
+    def test_aggregation_default_group_by(self):
+        qs = (
+            Publisher.objects.values('name')
+            .annotate(books=Count('book'), pages=Sum('book__pages', default=0))
+            .filter(books=0)
+        )
+        self.assertEqual(
+            list(qs),
+            [{'name': "Jonno's House of Books", 'books': 0, 'pages': 0}],
+        )
+
+    def test_aggregation_default_compound_expression(self):
+        # Scale rating to a percentage; default to 50% if no books published.
+        formula = Avg('book__rating', default=2.5) * 20.0
+        queryset = Publisher.objects.annotate(rating=formula).order_by('name')
+        self.assertQuerysetEqual(queryset, [
+            ('Apress', 85.0),
+            ("Jonno's House of Books", 50.0),
+            ('Morgan Kaufmann', 100.0),
+            ('Prentice Hall', 80.0),
+            ('Sams', 60.0),
+        ], lambda obj: (obj.name, obj.rating))
```

Comment:
We're trying to avoid `assertQuerysetEqual()` unless you really compare querysets
```suggestion
        self.assertSequenceEqual(queryset.values('name', 'rating'), [
            {'name': 'Apress', 'rating': 85.0},
            {'name': "Jonno's House of Books", 'rating': 50.0},
            {'name': 'Morgan Kaufmann', 'rating': 100.0},
            {'name': 'Prentice Hall', 'rating': 80.0},
            {'name': 'Sams', 'rating': 60.0},
        ])
```